include/ignition/gazebo/Server.hh

```diff
@@ -218,6 +219,16 @@ namespace ignition
                   const SystemPluginPtr &_system,
                   const unsigned int _worldIndex = 0);
 
+      /// \brief Add a System to the server. The server must not be running when
+      /// calling this. The server does not take ownership of the pointer and
+      /// assumes that the pointer is valid during the server's lifetime.
+      /// \param[in] _system system to be added
+      /// \param[in] _worldIndex Index of the world to query.
+      /// \return Whether the system was added successfully, or std::nullopt
+      /// if _worldIndex is invalid.
+      public: std::optional<bool> AddSystem(System *_system,
```

Comment:
Not really feedback on this PR, but I just realized that `_worldIndex` is used to index into an array of `SimulationRunner`s, which implies that there's a 1:1 relationship between worlds and `SimulationRunner`s. And since each `SimulationRunner` has its own `EntityComponentManager`, there can only be one world in an `EntityComponentManager`.

Reply:
Yeah this was documented a while back: https://ignitionrobotics.org/api/gazebo/5.1/terminology.html

> * Simulation runner: Runs a whole world or some levels of a world, but no more than 1 world.
>      * It has a single ECM with all the entities and components relevant to the levels / world / performer being simulated.
>           * It's still TBD how to support multiple `<worlds>` in parallel.

5 major versions later, and it's still TBD :grimacing: